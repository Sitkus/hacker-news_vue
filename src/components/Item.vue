<template>
  <li class="item">
    <h2 class="item__title">
      <a :href="item.url" target="_blank" rel="noopener" class="item__link">
        {{ item.title }}
        <span class="item__title-url"> ({{ item.url | host }}) </span>
      </a>
    </h2>

    <div class="meta">
      <span class="meta__score">{{ item.score }}</span>
      <span v-if="item.type !== 'job'" class="meta__by">
        by <router-link :to="'/user/' + item.by">{{ item.by }}</router-link>
      </span>
      <span v-if="item.type !== 'job'" class="meta__comments-link">
        | <router-link :to="'/item/' + item.id">{{ item.descendants }} comments</router-link>
      </span>
      <span> {{ item.time | timeAgo }} ago </span>
    </div>
  </li>
</template>

<script>
export default {
  name: 'Item',
  props: {
    item: {
      type: Object,
      required: true
    }
  }
};
</script>

<style scoped lang="scss">
.item {
  margin-bottom: 1rem;
  padding: 0.5rem 1.5rem;

  &__title {
    font-weight: 500;
    font-size: 1.6rem;
  }

  &__title-url {
    color: grey;
    font-size: 1.4rem;
    font-weight: 400;
  }
}

.meta {
  color: grey;
  font-size: 1.2rem;
}
</style>
