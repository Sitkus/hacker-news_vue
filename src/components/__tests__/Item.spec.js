import { shallowMount } from '@vue/test-utils';
import Item from '../Item.vue';

describe('Item.vue', () => {
  test('render a URL, an author and a score from item prop', () => {
    const item = {
      url: 'https://google.com',
      author: 'Bill',
      score: 20
    };

    const wrapper = shallowMount(Item, {
      propsData: { item }
    });

    expect(wrapper.text()).toContain(item.url);
    expect(wrapper.text()).toContain(item.author);
    expect(wrapper.text()).toContain(item.score);
  });

  test('render a link to the item.url with item.title as text', () => {
    const item = {
      url: 'https://google.com',
      title: 'some title'
    };

    const wrapper = shallowMount(Item, {
      propsData: { item }
    });

    const link = wrapper.find('a');

    expect(link.text()).toBe(item.title);
    expect(link.attributes().href).toBe(item.url);
  });
});
