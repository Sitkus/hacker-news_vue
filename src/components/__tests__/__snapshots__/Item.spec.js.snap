// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item.vue snapshots renders correctly 1`] = `
<li
  class="item"
>
  <h2
    class="item__title"
  >
    <a
      class="item__link"
      href="https://vue-test-utils.vuejs.org/"
      rel="noopener"
      target="_blank"
    >
      
      vue-test-utils is released
      
      <span
        class="item__title-url"
      >
         (vue-test-utils.vuejs.org) 
      </span>
    </a>
  </h2>
   
  <div
    class="meta"
  >
    <span
      class="meta__score"
    >
      10
    </span>
     
    <span
      class="meta__by"
    >
      
      by 
      <a>
        Lukas Sitkus
      </a>
    </span>
     
    <span
      class="meta__comments-link"
    >
      
      | 
      <a>
         comments
      </a>
    </span>
     
    <span>
       10 minutes ago 
    </span>
  </div>
</li>
`;

exports[`Item.vue snapshots renders correctly as job 1`] = `
<li
  class="item"
>
  <h2
    class="item__title"
  >
    <a
      class="item__link"
      rel="noopener"
      target="_blank"
    >
      
      vue-test-utils is released
      
      <span
        class="item__title-url"
      >
         () 
      </span>
    </a>
  </h2>
   
  <div
    class="meta"
  >
    <span
      class="meta__score"
    >
      10
    </span>
     
    <!---->
     
    <!---->
     
    <span>
       10 minutes ago 
    </span>
  </div>
</li>
`;
